tests/datasets/test_streaming.py

```diff
@@ -330,3 +358,35 @@ def test_dataloader_multi_device(remote_local: Tuple[str, str], batch_size: int,
         assert sample_order != second_sample_order
     else:
         assert sample_order == second_sample_order
+
+
+@pytest.mark.timeout(10)
+@pytest.mark.parametrize('compression', [None, 'gz', 'gz:5'])
+def test_compression(compressed_remote_local: Tuple[str, str, str], compression: Optional[str]):
+    num_samples = 31
+    shard_size_limit = 1 << 6
+    shuffle = True
+    compressed, remote, local = compressed_remote_local
+    samples, decoders = get_fake_samples_decoders(num_samples)
+
+    write_synthetic_streaming_dataset(dirname=compressed,
+                                      samples=samples,
+                                      shard_size_limit=shard_size_limit,
+                                      compression=compression)
+    write_synthetic_streaming_dataset(dirname=remote,
+                                      samples=samples,
+                                      shard_size_limit=shard_size_limit,
+                                      compression=None)
+
+    dataset = StreamingDataset(remote=compressed, local=local, shuffle=shuffle, decoders=decoders)
+
+    list(x for x in dataset)  # download samples
```

Comment:
The comment here is super helpful. Would prefer to iterate over it explicitly rather than convert it into a list (nit)
```suggestion
    for x in dataset:
        pass  # download sample
```